app/databaseAccessors/userAccessor.js

```diff
@@ -51,6 +51,54 @@ export default class UsersAccessor {
     }
   }
 
+  /**
+   * getApprovedByUsername Method
+   *
+   * This method retrieves the user MongoDB object from the
+   * database based on a given username
+   *
+   * @param {String} username
+   * @returns the User associated with the given username in
+   * the database.
+   */
+  static async getApprovedByUsername(username) {
+    try {
+      await Connection.open();
+      const user = await User.findOne({ username: username });
+      if (user && user.status === "approved") {
+        return user;
+      }
+      return null;
+    } catch (e) {
+      console.log(e);
+      ErrorInternalAPIModelFieldValidation(e);
```

Comment:
Not sure this is the appropriate error to throw here. Also, internal errors need keyword `throw` before them unlike HTTP errors.

The error here seems more like an unexpected error, unless Connection.open() throws a DB connection error. I would catch the error, see if it is an instance of the internal database connection error, and if so, throw it up the stack, if not, throw unexpected (because at that point we don't actually know what went wrong)

Also don't forget to import the errors.